hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java

```diff
@@ -41,47 +40,64 @@
 
     public Config overwriteMemberConfig(Config config) {
-        return overwrite(config, (provider, c) -> {
+        return overwrite(config, (provider, rootNode, target) -> {
               try {
-                  new YamlMemberDomConfigProcessor(true, c, false)
-                    .buildConfig(new ConfigOverrideElementAdapter(propsToNode(provider.properties())));
+                  new YamlMemberDomConfigProcessor(true, target, false)
+                    .buildConfig(new ConfigOverrideElementAdapter(rootNode));
               } catch (Exception e) {
-                  throw new InvalidConfigurationException("failed to overwrite configuration coming from " + provider.name(), e);
+                  throw new InvalidConfigurationException("failed to overwrite configuration coming from " + provider, e);
               }
           },
           new EnvConfigProvider(EnvVariablesConfigParser.member()),
           new SystemPropertiesConfigProvider(SystemPropertiesConfigParser.member()));
     }
 
     public ClientConfig overwriteClientConfig(ClientConfig config) {
-        return overwrite(config, (provider, c) -> {
+        return overwrite(config, (provider, rootNode, target) -> {
               try {
-                  new YamlClientDomConfigProcessor(true, c, false)
-                    .buildConfig(new ConfigOverrideElementAdapter(propsToNode(provider.properties())));
+                  new YamlClientDomConfigProcessor(true, target, false)
+                    .buildConfig(new ConfigOverrideElementAdapter(rootNode));
               } catch (Exception e) {
-                  throw new InvalidConfigurationException("failed to overwrite configuration coming from " + provider.name(), e);
+                  throw new InvalidConfigurationException("failed to overwrite configuration coming from " + provider, e);
               }
           },
           new EnvConfigProvider(EnvVariablesConfigParser.client()),
           new SystemPropertiesConfigProvider(SystemPropertiesConfigParser.client()));
     }
 
-    private <T> T overwrite(T config, BiConsumer<ConfigProvider, T> configProcessor, ConfigProvider... providers) {
+    private <T> T overwrite(T config, ConfigConsumer<T> configProcessor, ConfigProvider... providers) {
         ConfigOverrideValidator.validate(new HashSet<>(Arrays.asList(providers)));
 
         for (ConfigProvider configProvider : providers) {
             Map<String, String> properties = configProvider.properties();
 
             if (!properties.isEmpty()) {
-                LOGGER.info(format("Detected external configuration overrides in %s: [%s]",
+
+                ConfigNode rootNode = propsToNode(properties);
+                configProcessor.apply(configProvider.name(), rootNode, config);
+
+                Map<String, String> unprocessed = new ConfigNodeStateTracker().unprocessedNodes(rootNode);
+
+                LOGGER.info(format("Detected external configuration entries in %s: [%s]",
```

Comment:
In the edge case when `properties.size() == unprocessed.size()` I'd consider logging at WARN with a specific message saying that there are override properties defined, but we did not apply any of them.

Reply:
I'm not convinced. That would provide some extra value only for huge sets of config entries provided.

@Holmistr what do you think is the way to go?

Reply:
I don't see that much value in the extra log line. In general, I would expect the end user to go one by one when there's such warning. It's in his/her interest to fix those inconsistencies. Let's leave it like this.

Reply:
What I wanted to highlight is that in the mentioned case we would log "Detected external configuration entries in environment variables: []". With the warnings logged below the situation is clear, so I'm fine with the current approach as well.

Reply:
What if we just skip the empty log if the list if empty?

Reply:
I think it is reasonable to keep a log line that makes clear that there were no overrides. I would either leave this as is or log a specific message. I'm fine with both.